In [44]:
import numpy as np
import pandas as pd
import datetime

In [2]:
p1_gen = pd.read_csv("Plant_1/Plant_1_Generation_Data.csv")
p1_weath = pd.read_csv("Plant_1/Plant_1_Weather_Sensor_Data.csv")

p2_gen = pd.read_csv("Plant_2/Plant_2_Generation_Data.csv")
p2_weath = pd.read_csv("Plant_2/Plant_2_Weather_Sensor_Data.csv")

In [10]:
for df in [p1_gen, p1_weath, p2_gen, p2_weath]:
    df['DATE_TIME'] = pd.to_datetime(df['DATE_TIME'])

In [127]:
df_1g = p1_gen.copy()
df_1w = p1_weath.copy()
df_2g = p2_gen.copy()
df_2w = p2_weath.copy()

#for df in [df_1g, df_1w, df_2g, df_2w]:
df_1g = df_1g[ (df_1g['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_1g['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]
df_1w = df_1w[ (df_1w['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_1w['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]
df_2g = df_2g[ (df_2g['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_2g['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]
df_2w = df_2w[ (df_2w['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_2w['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]

In [137]:
df_2g

DATE_TIME  PLANT_ID       SOURCE_KEY   DC_POWER   AC_POWER  \
528   2020-05-15 06:00:00   4136001  4UPUqMRk7TRMgml  15.406667  14.860000   
529   2020-05-15 06:00:00   4136001  81aHJ1q11NBPMrL  14.773333  14.246667   
530   2020-05-15 06:00:00   4136001  9kRcWv60rDACzjR  15.380000  14.840000   
531   2020-05-15 06:00:00   4136001  Et9kgGMDl729KT4  14.973333  14.440000   
532   2020-05-15 06:00:00   4136001  IQ2d7wF4YD8zU1Q  15.160000  14.633333   
...                   ...       ...              ...        ...        ...   
67187 2020-06-17 18:00:00   4136001  q49J1IKaHRwDQnt  60.806667  58.886667   
67188 2020-06-17 18:00:00   4136001  rrq4fwE8jgrTyWY  57.420000  55.566667   
67189 2020-06-17 18:00:00   4136001  vOuJvMaM2sgwLmb  62.120000  60.193333   
67190 2020-06-17 18:00:00   4136001  xMbIugepa2P7lBB  61.420000  59.480000   
67191 2020-06-17 18:00:00   4136001  xoJJ8DcxJEcupym  61.393333  59.453333   

       DAILY_YIELD   TOTAL_YIELD  
528       0.800000  2.429012e+06  
529       0.733333  1.215279e+09  
530       1.866667  2.247720e+09  
531       1.000000  1.704251e+06  
532       0.733333  1.994153e+07  
...            ...           ...  
67187  4136.200000  5.207372e+05  
67188  3911.466667  1.211313e+08  
67189  4301.066667  2.427670e+06  
67190  4197.333333  1.068964e+08  
67191  4295.066667  2.093357e+08  

[34506 rows x 7 columns]

In [138]:
gen_zeros = []
for df in [df_1g, df_2g]:
    DC = (df['DC_POWER'] == 0).sum()
    AC = (df['AC_POWER'] == 0).sum()
    DY = (df['DAILY_YIELD'] == 0).sum()
    TY = (df['TOTAL_YIELD'] == 0).sum()
    zeros = DC + AC + DY + TY
    gen_zeros.append(zeros)
    
weath_zeros = []
for df in [df_1w, df_2w]:
    amb = (df['AMBIENT_TEMPERATURE'] == 0).sum()
    mod = (df['MODULE_TEMPERATURE'] == 0).sum()
    irr = (df['IRRADIATION'] == 0).sum()
    zeros = amb + mod + irr
    weath_zeros.append(zeros)

In [139]:
print(gen_zeros)
print(weath_zeros)

[709, 9154]
[0, 1]


##### Plant 1 zeros
Gen = 709, all = 143 552

Weather = 0, all = 4905

Together = 709/148 457 = 0.005

##### Plant 2 zeros
Gen = 9154, all = 138 024

Weather = 1, all = 4986

Together = 9155/143 010 = 0.64